src/NonEmptyArray2v.ts

```diff
@@ -281,6 +282,30 @@ const mapWithIndex = <A, B>(fa: NonEmptyArray<A>, f: (i: number, a: A) => B): No
   return fa.map((a, i) => f(i, a))
 }
 
+/**
+ * Append an element to the end of an non empty array, creating a new non empty array
+ *
+ * @example
+ * import { snoc } from 'fp-ts/lib/NonEmptyArray2v'
+ *
+ * assert.deepStrictEqual(snoc(fromNonEmptyArray([1, 2, 3]), 4), [1, 2, 3, 4])
+ *
+ * @since 1.16.0
+ */
+export const snoc: <A>(as: NonEmptyArray<A>, a: A) => NonEmptyArray<A> = array.snoc as any
```

Comment:
why not `<A>(as: Array<A>, a: A) => NonEmptyArray<A>`?

Reply:
@sledorze thinking twice... maybe we should just change `cons` / `snoc` signatures in the `Array` module instead of duplicating

Reply:
Oh yes that's true, now we have nonemptyarray2v :)

Reply:
Maybe we could keep some methods in nonemptyarray2v to help with DX.

Reply:
> to help with DX

Good point.

btw there's already a `import * as A from './Array'` at the beginning of the file